test/test_eos_unit.cpp

```diff
@@ -509,6 +509,71 @@ SCENARIO("Ideal gas entropy", "[IdealGas][Entropy]") {
   }
 }
 
+// A non-standard pattern where we do a reduction
+class CheckPofRE {
+ public:
+  CheckPofRE(Real *P, Real *rho, Real *sie, int N) : P_(P), rho_(rho), sie_(sie), N_(N) {}
+  template <typename T>
+  void operator()(const T &eos) {
+    portableReduce(
+        "MyCheckPofRE", 0, N_,
+        PORTABLE_LAMBDA(const int i, int &nw) {
+          nw += !(isClose(P_[i],
+                          eos.PressureFromDensityInternalEnergy(rho_[i], sie_[i], nullptr),
+                          1e-15));
+        },
+        nwrong);
+  }
+
+  // must be initialized to zero because portableReduce is a simple for loop on host
+  int nwrong = 0;
+
+ private:
+  int N_;
+  Real *P_;
+  Real *rho_;
+  Real *sie_;
+};
+SCENARIO("Ideal gas vector Evaluate call", "[IdealGas][Evaluate]") {
```

Comment:
Story for later, but I really want to split up `test_eos_unit.cpp` into more, smaller files. Some day soon.